addons/controller_icons/objects/Button.gd

```diff
@@ -1,27 +1,26 @@
-tool
+@tool
 extends Button
 class_name ControllerButton
 
-export(String) var path : String = "" setget set_path
-export(int, "Both", "Keyboard/Mouse", "Controller") var show_only : int = 0 setget set_show_only
+@export var path : String = "":
+	set(_path):
+		path = _path
+		if is_inside_tree():
+			icon = ControllerIcons.parse_path(path)
+			
+@export_enum("Both", "Keyboard/Mouse", "Controller") var show_only : int = 0:
+	set(_show_only):
+		show_only = _show_only
+		_on_input_type_changed(ControllerIcons._last_input_type)
 
 func _ready():
-	ControllerIcons.connect("input_type_changed", self, "_on_input_type_changed")
-	set_path(path)
+	ControllerIcons.input_type_changed.connect(_on_input_type_changed)
+	path = path
```

Comment:
This can be confusing. A reference to the instance variable looks better
```suggestion
	self.path = path
```
Not sure how this changed for Godot 4, but for Godot 3 it was like this:
```gdscript
.path = path
```
This should be done for the other objects too (Sprite2D, Sprite3D and TextureRect)